# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/TSFTextStore.h
# Commit: 14e68c94b303
# Full Hash: 14e68c94b3036427ed9cac5adca7615b11d1df81
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 1-1: Make `TSFTextStore::Selection` work with `IMENotification::SelectionChangeDataBase` directly r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137404
# ==============================================================================

diff -r 41535e3d6028 -r 14e68c94b303 widget/windows/TSFTextStore.h
--- a/widget/windows/TSFTextStore.h	Mon Feb 07 22:27:52 2022 +0000
+++ b/widget/windows/TSFTextStore.h	Mon Feb 07 22:33:30 2022 +0000
@@ -555,6 +555,10 @@
       Collapse(aOffsetToCollapse);
     }
 
+    explicit Selection(const SelectionChangeDataBase& aSelectionChangeData) {
+      SetSelection(aSelectionChangeData);
+    }
+
     explicit Selection(uint32_t aStart, uint32_t aLength, bool aReversed,
                        const WritingMode& aWritingMode) {
       SetSelection(aStart, aLength, aReversed, aWritingMode);
@@ -571,6 +575,13 @@
       mACP.style.fInterimChar = FALSE;
     }
 
+    bool SetSelection(const SelectionChangeDataBase& aSelectionChangeData) {
+      return SetSelection(aSelectionChangeData.mOffset,
+                          aSelectionChangeData.Length(),
+                          aSelectionChangeData.mReversed,
+                          aSelectionChangeData.GetWritingMode());
+    }
+
     bool SetSelection(uint32_t aStart, uint32_t aLength, bool aReversed,
                       const WritingMode& aWritingMode) {
       bool changed = mACP.acpStart != static_cast<LONG>(aStart) ||
